### DISCUSSION TITLE
Starting dap without having a Swift buffer open

### DISCUSSION THREAD

**Discussion by whiteio** [wojciech-kulik/xcodebuild.nvim#49] [Q&A] [answered]
Hey! Just had a small question about how the debugger starts. I noticed that a Swift buffer needed to be open to start dap within `start_dap_in_swift_buffer`. Was just curious if it'd be able to start with a different buffer open such as Objective-C?

**Comment by wojciech-kulik** [comment, marked as answer]
This is how `dap.continue()` function works. It gets the DAP configuration based on the current file type. The function `start_dap_in_swift_buffer` calls `dap.continue()` from the first Swift buffer it finds, without switching to it by using `vim.api.nvim_win_call`.

> Was just curious if it'd be able to start with a different buffer open such as Objective-C?

As long as there is any Swift buffer visible.

**Comment by wojciech-kulik** [comment]
@whiteio Actually, it turned out that I can workaround it by using `dap.run(dap.configurations.swift)` instead of `dap.continue()`. This way you don't have to have any Swift buffer open.

Thank you for the feedback it made me think about this solution and as a result fixing the problem :)! 🍻 Please update the plugin and enjoy!
